EKF/control.cpp

```diff
@@ -1538,16 +1538,10 @@ void Ekf::controlMagFusion()
 			}
 
 		} else if (_params.mag_fusion_type == MAG_FUSE_TYPE_3D) {
-			// if transitioning into 3-axis fusion mode, we need to initialise the yaw angle and field states
-			if (!_control_status.flags.mag_3D || !_control_status.flags.mag_align_complete) {
-				_control_status.flags.mag_align_complete = resetMagHeading(_mag_sample_delayed.mag);
-				_control_status.flags.yaw_align = _control_status.flags.yaw_align || _control_status.flags.mag_align_complete;
+			if (!_control_status.flags.mag_3D && _control_status.flags.yaw_align && (_flt_mag_align_start_time > 0)) {
```

Comment:
Why do you need to check for `_flt_mag_align_start_time > 0` here?

Reply:
@priseborough

Reply:
Originally it was to prevent the use of 3-axis mag fusion when on the ground whichcan casue problems due to ground based magnetic anomalies.

Reply:
`_flt_mag_align_start_time` is set to the [imu time on bootup](https://github.com/PX4/ecl/blob/cd2e0739c6151ab9f444fd5ec190c89610db8279/EKF/ekf_helper.cpp#L713-L713) the [first time we trigger resetMagheading()](https://github.com/PX4/ecl/blob/cd2e0739c6151ab9f444fd5ec190c89610db8279/EKF/ekf.cpp#L265-L265) so the check here will never be false

Reply:
I'll fix that and rebase.